### ISSUE TITLE
selecting text is terrible and buggy

### ISSUE THREAD

**Issue by um6reon** [maelchiotti/LocalMaterialNotes#160]
sometimes after failing to select a text or copy & paste, the selected text can't be deleted.. instead the app decides to delete the text before it, with the char lenght of the selected text?

**Comment by maelchiotti** [comment]
Hi @raelite, I don't encounter such an issue on my device, so it's hard for me to find with this occurs. Does it happen often?

**Comment by raelite** [comment]
> Hi @raelite, I don't encounter such an issue on my device, so it's hard for me to find out why this occurs. Does it happen often?

it does happen often yes, almost every time.
i dont think there is anything i could provide or test any further?

doesn't happen for other notes apps etc.

**Comment by raelite** [comment]
@maelchiotti  does this help? selecting text, tapping on an empty spot, selection again -> delete?

https://github.com/user-attachments/assets/66ed4157-ba80-4e50-9670-fc75d7617da7

**Comment by maelchiotti** [comment]
No, it works fine both with my physical device and in my emulator... I can open an issue in the [fleather](https://github.com/fleather-editor/fleather) package (used for the note editor) to see if they have ever encountered this.

**Comment by raelite** [comment]
> No, it works fine both with my physical device and in my emulator... I can open an issue in the [fleather](https://github.com/fleather-editor/fleather) package (used for the note editor) to see if they have ever encountered this.

had the idea earlier but was to lazy to test it.. and wow.. i should've done it earlier! Soory :)

so after you said for ya it works.. i did try a different keyboard app besides 'HeliBoard' now and well.. it is the combination of HeliBoard AND Material Notes.. the keyboad doesn't seem to like your app? or fleather? fossify keyboard and floris etc all do work with ya app, its just heliboard on your specific app..? Weeeird @.@

**Comment by AstroSquared** [comment]
I can reproduce this with the FUTO keyboard and Material Notes 1.5.2. It's even buggier then just selected text replacement and can type/delete things where the cursor was previously until you click on the current cursor location again.

**Comment by maelchiotti** [comment]
Oh indeed, I just tried it out and it's really easy to reproduce the issue. I'll ask the fleather team about it, because the title text field which is a regular Flutter widget works fine.
